crates/types/src/epoch.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2024 - Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+#![allow(dead_code)]
+
+use crate::identifiers::{LeaderEpoch, PartitionId};
+use crate::{GenerationalNodeId, Version, Versioned};
+
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Epoch {
```

Comment:
nit: maybe "EpochMetadata" to make the distinction between the type we use for epochs (as in epoch of an ESN) vs. the metadata about an epoch?

Reply:
Good point. Will change it.